### ISSUE TITLE
sync: TestMutexFairness flaky on openbsd-*-62

### ISSUE THREAD

**Issue by bcmills** [golang/go#35281]
Looks like `TestMutexFairness` has somehow recently regressed on OpenBSD 6.2:

https://build.golang.org/log/1072b7783fa59d0db61b3dc39f893330491a7722
https://build.golang.org/log/4978ed9696cf9aed6c6e26c4fa7cb97ad55c99a7
https://build.golang.org/log/8bfbe3f24eb9dc9cf8b5f223130c208e2c786892
https://build.golang.org/log/5dd265480a37db6270267bc25822de1a62e04163

CC @bradfitz @aclements

**Comment by danscales** [comment]
I was able to reproduce this reliably at the change 02a5502ab8d862309aaec3c5ec293b57b913d01d listed in one of the failure instances, by repeating this 100 times:

gomote run  user-danscales-openbsd-amd64-62-0 go/bin/go test sync -test.run TestMutexFairness -test.count=1 -test.timeout=15s

Using this same way to reproduce, I found that it started happening when Ian turned on the new timer code.  It is fixed (not reproducible) in the current master, and by bisecting, I found it was fixed by Ian's commit 

d80ab3e85a :  runtime: wake netpoller when dropping P, don't sleep too long in sysmon

@ianlancetaylor  does that make sense to you (seems reasonable to me based on the description of the change and the bugs that it fixed).  If so, I think we can close this bug.

(Also, doesn't seem like it has been happening on the builder for the last week, as expected.)
